csvcubed/csvcubed/models/codelistconfig/code_list_config.py

```diff
@@ -0,0 +1,174 @@
+        """
+        Sorting concepts based on the sort object and sort order defined in the code list json.
+        """
+        # If the sort object is not defined, the sorting will default to the sort_order where defined.
+        if self.sort is None:
+            sorted_concepts: List[CodeListConfigConcept] = sorted(
+                concepts,
+                key=lambda x: (x.sort_order is None, x.sort_order),
+                reverse=False,
+            )
+            return sorted_concepts
+
+        # Otherwise, the sort object and sort_order both are used for sorting.
+        if self.sort.by != "label" and self.sort.by != "notation":
+            raise Exception(
+                f"Unsupported sort by {self.sort.by}. The supported options are 'label' and 'notation'."
+            )
+        if self.sort.method != "ascending" and self.sort.by != "descending":
+            raise Exception(
+                f"Unsupported sort method {self.sort.method}. The supported options are 'ascending' and 'descending'."
+            )
+
+        sorted_concepts: List[CodeListConfigConcept] = sorted(
+            concepts,
+            key=lambda x: (
+                x.sort_order is None,
+                x.sort_order,
+                x.label if self.sort and self.sort.by == "label" else x.notation,
+            ),
+            reverse=True if self.sort.method == "descending" else False,
+        )
+        # Assigning the new sort_order to the concepts.
+        for idx, concept in enumerate(sorted_concepts):
+            concept.sort_order = idx
+
+        return sorted_concepts
+
+    @classmethod
+    def from_json_file(cls, file_path: Path) -> Tuple["CodeListConfig", Dict]:
+        """
+        Converts code list config json to `CodeListConfig`.
+        """
+        code_list_dict = load_json_document(file_path)
+        schema = code_list_dict.get("$schema", CODE_LIST_CONFIG_DEFAULT_URL)
+
+        code_list_config = cls.from_dict(code_list_dict)
+        code_list_config.schema = schema
+        code_list_config.metadata = metadata_from_dict(code_list_dict)
+
+        return (code_list_config, code_list_dict)
+
+    @classmethod
+    def from_dict(cls, code_list_dict: Dict) -> "CodeListConfig":
+        """
+        Converts code list config dict to `CodeListConfig`.
+        """
+        schema = code_list_dict.get("$schema", CODE_LIST_CONFIG_DEFAULT_URL)
+
+        code_list_config = super().from_dict(code_list_dict)
+        code_list_config.schema = schema
+        code_list_config.metadata = metadata_from_dict(code_list_dict)
+
+        return code_list_config
+
+    @property
+    def new_qb_concepts(self) -> list[NewQbConcept]:
+        """
+        Converts concepts of type CodeListConfigConcept to concepts of type NewQbConcept whilst maintaining the hierarchy.
+        """
+
+        new_qb_concepts: list[NewQbConcept] = []
+        if self.concepts:
+            concepts_with_maybe_parent: list[
```

Comment:
I hope you don't mind but I've removed the funciton which looks up the parent concept and replaced it with something shorter and more performant.